vreplgen/vreplgen.go

```diff
@@ -23,35 +23,79 @@ package main
 
 import (
 	"bytes"
+	"flag"
 	"fmt"
+	"os"
 	"strings"
 
 	"vitess.io/vitess/go/sqltypes"
 	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
 )
 
+var onDdl string
+
+func init() {
+	flag.StringVar(&onDdl, "on_ddl", "ignore", "Set on_ddl value for replication stream - ignore, stop, exec, exec_ignore")
+	flag.Parse()
+}
+
 func main() {
-	vtctl := "./lvtctl.sh"
-	tabletID := "test-400"
-	dbName := "vt_merchant"
+	argOffset := 0
+	if (len(os.Args) > 1 && strings.HasPrefix(os.Args[1], "-")) {
+		argOffset = 2
+	}
+
+	if (len(os.Args) < (7+argOffset)) {
+		fmt.Println("Usage: vreplgen [-on_ddl (ignore|stop|exec|exec_ignore)] <tablet_id> <src_keyspace> <src_shard> <dest_keyspace> <dest_table1> 'filter1' [<dest_table2> 'filter2']...")
+		os.Exit(1)
+	}
+
+	vtctl := os.Getenv("VTCTLCLIENT")
+	if (vtctl == "") {
+	  vtctl = "vtctlclient -server localhost:15999"
+	}
+	tabletID := os.Args[1+argOffset]
+	sourceKeyspace := os.Args[2+argOffset]
+	sourceShard := os.Args[3+argOffset]
+	destKeyspace := os.Args[4+argOffset]
+	destDbName := "vt_" + destKeyspace
+	var rules []*binlogdatapb.Rule
+	for i := 5+argOffset; i < len(os.Args); i = i+2 {
+		destTable := os.Args[i]
+		destFilter := os.Args[i+1]
+		rule := new(binlogdatapb.Rule)
+		rule.Match = destTable
+		rule.Filter = destFilter
+		rules = append(rules, rule)
+	}
 	filter := &binlogdatapb.Filter{
-		Rules: []*binlogdatapb.Rule{{
-			Match:  "morder",
-			Filter: "select * from uorder where in_keyrange(mname, 'unicode_loose_md5', '-80')",
-		}},
+		Rules: rules,
 	}
+
+	var onDdlAction binlogdatapb.OnDDLAction
```

Comment:
You can change this to: `onDDLAction := binlogdatapb.OnDDLAction_value[strings.ToUpper(onDDL)]`.
Note that go standards require `DDL` to be capitalized inside variables.